### ISSUE TITLE
Set up glossary for translations with Weblate

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#2442]
We currently use [Weblate](https://hosted.weblate.org/engage/openrefine/?utm_source=widget) to translate the user interface. Translators get a list of untranslated strings (or strings needing review) in their language, write translations, and this gets committed to the repository via regular pull requests.

Weblate makes it possible to set up a [glossary](https://docs.weblate.org/en/latest/user/translating.html#glossary), which determines how the terminology that is specific to OpenRefine should be translated in each language. For instance, terms like "facet", "clustering" or "reconciliation" have a pretty specific meaning in our context - so we want to make sure these terms are translated uniformly in all strings.

We should look into setting up such a glossary and identifying the terms that should go there.

**Comment by comradekingu** [comment]
You could add that as context to the strings in question.

**Comment by WaltonG** [comment]
@wetneb @antoine2711 I would like to work on this

**Comment by WaltonG** [comment]
I got the following words from weblate. Ill go ahead and try to define them in OpenRefine context

Apply
best candidate
blank
Blanks
Cancel
Change
claim
clipboard
Cluster
Clusters
Collapse
Collapse
Column
complete
Computer
Contributors
Creator
Custom
Custom Tabular Exporter
database
data set
Delete
description
Description
Download
Downloading
edit
English
Enter
entity
entity identifier
Error
extension
Facet
facets
Facets
Help
HTML table
Identical
Internal error
judgment
key column
Language
license
Locate
match
metadata
Name
note column
Open
Options
Parse
pasted
preview
Preview
project
qualifier
QuickStatements
reconcile
reconciliation
reconciling
regular expression
Remove
Schema:
Search
Show
Sort
specify
statement
Subject
tag
Tag
Tags
Title
try
Unknown error
Updating
Upload
Uploading
value column
worksheet

**Comment by antoine2711** [comment]
> I got the following words from weblate. I will go ahead and try to define them in OpenRefine context

@WaltonG: a short description is good, as there is a text field for explanations. The next step would be to import that in Weblate. Currently, there is TWO word defined in the glossary: QuickStatements and Facet.

Here's a snapshoot of the editing pane in the wep page of Weblate:
![image](https://user-images.githubusercontent.com/12217525/162352877-77253974-ee99-4c6e-9a84-8c81f2e6d37e.png)

Here's the french link to OpenRefine’s Weblate glossary:
https://hosted.weblate.org/translate/openrefine/glossary/fr/

Have you’ve been there?

Regards, Antoine

**Comment by WaltonG** [comment]
@antoine2711 Yes, that's where I got the words to be included in the glossary I am working on. Thank you for the pointers

**Comment by WaltonG** [comment]
@antoine2711 Since it will be imported to weblate I guess I will have to present it in json format ?

**Comment by antoine2711** [comment]
> @antoine2711 Since it will be imported to weblate I guess I will have to present it in json format ?

@WaltonG: from what I understand, we currently have 4 Weblate « composant » that get transfered in our GH repo in 4 different locations. And yes, they are in JSON. (Here, one composant in one language, french.)

https://github.com/OpenRefine/OpenRefine/blob/3d6960912f418e4bbee089771b7c9aae3d2177df/extensions/wikidata/module/langs/translation-fr.json#L1-L5

![image](https://user-images.githubusercontent.com/12217525/162359718-f7e3e4b2-224a-4cf4-9604-992ff6b3600b.png)

I guess the glossary, a 5th composant, could also be transfered in our repo? But where? Yours to find out and propose! ;-)

Regards, Antoine

**Comment by WaltonG** [comment]
> > @antoine2711 Since it will be imported to weblate I guess I will have to present it in json format ?
> 
> @WaltonG: from what I understand, we currently have 4 Weblate « composant » that get transfered in our GH repo in 4 different locations. And yes, they are in JSON. (Here, one composant in one language, french.)
> 
> https://github.com/OpenRefine/OpenRefine/blob/3d6960912f418e4bbee089771b7c9aae3d2177df/extensions/wikidata/module/langs/translation-fr.json#L1-L5
> 
> ![image](https://user-images.githubusercontent.com/12217525/162359718-f7e3e4b2-224a-4cf4-9604-992ff6b3600b.png)
> 
> I guess the glossary, a 5th composant, could also be transfered in our repo? But where? Yours to find out and propose! ;-)
> 
> Regards, Antoine

@antoine2711 My proposal is to add a glossary folder in docs

**Comment by wetneb** [comment]
Why do we need to have the glossary in the repo anyway? Can we not keep it in Weblate?

**Comment by trnstlntk** [comment]
After looking into it a bit more deeply on Weblate itself: I agree that the glossary needs to be maintained in Weblate through its own end-user interface.

Work on that glossary may need social/community support, rather than a technical solution (IMO it would for instance be good to have a forum / online space where translators discuss terms and workflows among each other). I suggest that we keep this issue open, for visibility, as getting a functioning glossary for translators remains a valid request.

**Comment by antoine2711** [comment]
> Why do we need to have the glossary in the repo anyway? Can we not keep it in Weblate?

@wetneb: In order to see it's evolution? Keep a history? I don't believe Weblate gives that to us.

Regards,
   Antoine

**Comment by wetneb** [comment]
As you both see fit!
